content/client-server-api/_index.md

```diff
@@ -2805,42 +2805,42 @@ operations and run in a resource constrained environment. Like embedded
 applications, they are not intended to be fully-fledged communication
 systems.
 
-{{< cs-module name="instant_messaging" >}}
```

Comment:
hrm. We switched to `{{<` deliberately (https://github.com/matrix-org/matrix-spec/pull/1205). Switching back seems like it will break things?

Reply:
Oh, this is ok because the `cs-module` shortcode is switching to `.RenderShortcodes` instead of `.Content`? But why is it necessary?

Reply:
Hugo is lazy and uses the markdown renderer to detect headings and construct the headings tree.

`.Content` returns a page rendered as HTML, and `{{<` means "don't process the data returned by this shortcode with the renderer", so the renderer in the main page just replaces the shortcode with the module page rendered as HTML without processing anything, and Hugo is only aware of the headings in the main page.

`.RenderShortcodes` returns the raw markdown, with the shortcodes of the page already processed, and `{{%` means "process the data returned by this shortcode with the renderer", so the renderer processes the full mardown of the main page and the module pages as a single document, so Hugo is aware of all the headings in the final page.